modules/apps/transfer/keeper/relay_forwarding_test.go

```diff
@@ -612,3 +613,169 @@ Test async ack is properly relayed to middle hop after forwarding transfer compl
+	suite.assertAmountOnChain(suite.chainB, escrow, amount, denomAB.IBCDenom(), "Chain B's escrow should have coins")
+
+	address := types.GetForwardAddress(packet.DestinationPort, packet.DestinationChannel).String()
+	data := types.NewFungibleTokenPacketDataV2(
+		[]types.Token{
+			{
+				Denom:  types.NewDenom(sdk.DefaultBondDenom, types.NewTrace(pathAtoB.EndpointA.ChannelConfig.PortID, pathAtoB.EndpointA.ChannelID)),
+				Amount: "100",
+			},
+		},
+		address,
+		suite.chainC.SenderAccounts[0].SenderAccount.GetAddress().String(),
+		"", nil,
+	)
+
+	packet = channeltypes.NewPacket(
+		data.GetBytes(),
+		1,
+		pathBtoC.EndpointA.ChannelConfig.PortID,
+		pathBtoC.EndpointA.ChannelID,
+		pathBtoC.EndpointB.ChannelConfig.PortID,
+		pathBtoC.EndpointB.ChannelID,
+		packet.TimeoutHeight,
+		packet.TimeoutTimestamp)
+
+	// retrieve module callbacks
+	module, _, err := suite.chainB.App.GetIBCKeeper().PortKeeper.LookupModuleByPort(suite.chainB.GetContext(), pathBtoC.EndpointA.ChannelConfig.PortID)
+	suite.Require().NoError(err)
+
+	cbs, ok := suite.chainB.App.GetIBCKeeper().PortKeeper.Route(module)
+	suite.Require().True(ok)
+
+	// Trigger OnTimeoutPacket for chainB
+	err = cbs.OnTimeoutPacket(suite.chainB.GetContext(), packet, nil)
+	suite.Require().NoError(err)
+
+	// Ensure that chainB has an ack.
+	res := suite.chainB.GetAcknowledgement(packet)
+	suite.Require().NotNil(res, "chainB does not have an ack")
+
+	// And that this ack is of the type we expect (Error due to time out)
+	ack := channeltypes.NewErrorAcknowledgement(errors.New("forwarded packet timed out"))
+	ackbytes := channeltypes.CommitAcknowledgement(ack.Acknowledgement())
+	suite.Require().Equal(ackbytes, res)
+
+	data = types.NewFungibleTokenPacketDataV2(
+		[]types.Token{
+			{
+				Denom:  types.NewDenom("stake"),
+				Amount: "100",
+			},
+		},
+		suite.chainA.SenderAccounts[0].SenderAccount.GetAddress().String(),
+		suite.chainC.SenderAccounts[0].SenderAccount.GetAddress().String(),
+		"", hops,
+	)
+
+	packet = channeltypes.NewPacket(
+		data.GetBytes(),
+		1,
+		pathAtoB.EndpointA.ChannelConfig.PortID,
+		pathAtoB.EndpointA.ChannelID,
+		pathAtoB.EndpointB.ChannelConfig.PortID,
+		pathAtoB.EndpointB.ChannelID,
+		packet.TimeoutHeight,
+		packet.TimeoutTimestamp)
+
+	// Send the ack to chain A.
+	err = suite.chainA.GetSimApp().TransferKeeper.OnAcknowledgementPacket(suite.chainA.GetContext(), packet, data, ack)
+	suite.Require().NoError(err)
+
+	// Finally, check that A,B, and C escrow accounts do not have fund.
+	suite.assertAmountOnChain(suite.chainB, escrow, sdkmath.NewInt(0), denomAB.IBCDenom(), "Escrow account for chain B should be empty")
+	suite.assertAmountOnChain(suite.chainC, escrow, sdkmath.NewInt(0), denomABC.IBCDenom(), "Escrow account for chain C should be empty")
+	suite.assertAmountOnChain(suite.chainA, escrow, sdkmath.NewInt(0), denomA.IBCDenom(), "Escrow account for chain a should be empty")
```

Comment:
I would just order them differently :) 

```suggestion
	suite.assertAmountOnChain(suite.chainC, escrow, sdkmath.NewInt(0), denomABC.IBCDenom(), "Escrow account for chain C should be empty")
	suite.assertAmountOnChain(suite.chainB, escrow, sdkmath.NewInt(0), denomAB.IBCDenom(), "Escrow account for chain B should be empty")
	suite.assertAmountOnChain(suite.chainA, escrow, sdkmath.NewInt(0), denomA.IBCDenom(), "Escrow account for chain a should be empty")
```